app/models/waste_carriers_engine/convictions_check/entity.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+module WasteCarriersEngine
+  module ConvictionsCheck
+    class Entity
+      include Mongoid::Document
+
+      store_in collection: "entities"
+
+      # The full name, either of an organisation or an individual
+      field :name,                                 type: String
+      # Only used for individuals - may not be known or available
+      field :dateOfBirth, as: :date_of_birth,      type: Date
+      # Only used for organisations, if applicable
+      field :companyNumber, as: :company_number,   type: String
+      # The system where the match is from
+      field :systemFlag, as: :system_flag,         type: String
+      # The incident number or reference code
+      field :incidentNumber, as: :incident_number, type: String
+
+      scope :matching_organisation_name, lambda { |term|
+        where(name: /#{org_name_search_term(term)}/i)
+      }
+
+      scope :matching_person_name, lambda { |first_name:, last_name:|
+        escaped_first_name = ::Regexp.escape(first_name) if first_name.present?
+        escaped_last_name = ::Regexp.escape(last_name) if last_name.present?
+
+        where(name: /#{escaped_first_name}/i).and(name: /#{escaped_last_name}/i)
+      }
+
+      scope :matching_date_of_birth, lambda { |term|
+        where(date_of_birth: term)
+      }
+
+      scope :matching_company_number, lambda { |term|
+        escaped_term = ::Regexp.escape(term) if term.present?
```

Comment:
shall we return if there is no term present? Line 41 will fail if the term is not present and we are missing test coverage for when a term is empty

Reply:
Good shout – I've added test coverage and set the scopes to raise an error if the terms aren't present!